File: app/src/main/java/com/guliash/countryquiz/core/BaseFragment.java
package com.guliash.countryquiz.core;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import timber.log.Timber;

public class BaseFragment extends Fragment {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        Timber.d("ON CREATE");
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onDestroy() {
        Timber.d("ON DESTROY");
        super.onDestroy();
    }

    @Override
    public void onAttach(Context context) {
        Timber.d("ON ATTACH");
        super.onAttach(context);
    }

    @Override
    public void onDetach() {
        Timber.d("ON DETACH");
        super.onDetach();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        Timber.d("ON CREATE VIEW");
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onDestroyView() {
        Timber.d("ON DESTROY VIEW");
        super.onDestroyView();
    }

    @Override
    public void onStart() {
        Timber.d("ON START");
        super.onStart();
    }

    @Override
    public void onStop() {
        Timber.d("ON STOP");
        super.onStop();
    }

    @Override
    public void onResume() {
        Timber.d("ON RESUME");
        super.onResume();
    }

    @Override
    public void onPause() {
        Timber.d("ON PAUSE");
        super.onPause();
    }
}

File: app/src/main/java/com/guliash/quizzes/core/ui/fragment/BaseFragment.kt
package com.guliash.quizzes.core.ui.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View

open class BaseFragment : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        println("${this}: onCreate")
        super.onCreate(savedInstanceState)
    }

    override fun onDestroy() {
        println("${this}: onDestroy")
        super.onDestroy()
    }

    override fun onStart() {
        println("${this}: onStart")
        super.onStart()
    }

    override fun onStop() {
        println("${this}: onStop")
        super.onStop()
    }

    override fun onResume() {
        println("${this}: onResume")
        super.onResume()
    }

    override fun onPause() {
        println("${this}: onPause")
        super.onPause()
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        println("${this}: onViewCreated")
        super.onViewCreated(view, savedInstanceState)
    }

    override fun onDestroyView() {
        println("${this}: onDestroyView")
        super.onDestroyView()
    }

    override fun onViewStateRestored(savedInstanceState: Bundle?) {
        println("${this}: onViewStateRestored")
        super.onViewStateRestored(savedInstanceState)
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        println("${this}: onSaveInstanceState")
        super.onSaveInstanceState(outState)
    }
}